crates/bevy_math/src/aspect_ratio.rs

```diff
@@ -10,18 +10,81 @@ use bevy_reflect::Reflect;
 #[cfg_attr(feature = "bevy_reflect", derive(Reflect), reflect(Debug, PartialEq))]
 pub struct AspectRatio(f32);
 
+#[derive(Debug, PartialEq)]
+pub enum AspectRatioError {
```

Comment:
We should probably make this into a full blown error type. I'd advise splitting this out into its own module, so not to clutter this file with the needed `impl`s to get it to work.

You can look at https://github.com/bevyengine/bevy/blob/main/crates/bevy_ecs/src/identifier/error.rs as an example of this.

So we'll need at least `Clone`, `Copy`, `Eq` as well derived, then implement `std::fmt::Display` for `AspectRatioError`. After that, you can implement `std::error::Error`.

Reply:
I tried to keep the overall design the same as the other bevy crates.